### PR TITLE
Split role description and application form

### PR DESCRIPTION
**Pull request by mtruj013**
## Done

- Separated application form and job description 

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: https://canonical-com-759.demos.haus/careers/all
- Click on any job posting on and see that you only see the job description
- Scroll down and click "Apply" and see that you are taken to the form and that the url now has the `/application` in the path
- At the top of the page click "Back to role description" and see that the form is once again hidden and the url does not have `/aplication` in the path 

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-958

### CODE REVIEW

**Comment by lyubomir-popov** [comment]
@mtruj013 can we please do this on both the text and form page?
image

- add `p-heading--2` to h1
- - add `p-heading--5` to the h4s
- split layout on desktop, as is one smaller screens

File: templates/careers/job-detail.html [hunk 1]
```diff
@@ -82,13 +80,18 @@ <h1>{{ job.title }}</h1>
 </div>
 {% endif %}
 <div class="row">
-  <div class="col-6">
+  <div class="col-8">
     <div class="job-desc">
       {{ job.content | filtered_html_tags | safe }}
     </div>
+    <p><a class="p-button--positive" id="apply-button">Apply</a></p>
```
Comment (anthonydillon) [inline templates/careers/job-detail.html:87]: This is invalid mark up. Links need to have a `href` attribute

File: templates/careers/base_job-details.html [hunk 2]
```diff
@@ -2,8 +2,44 @@
 
 {% block hero %}{% endblock %}
 
-<section class="p-strip u-extra-space">
+<section class="p-strip u-extra-space" id="details">
   {% block careers_content %}{% endblock %}
 </section>
+<section class="p-strip u-extra-space u-hide" id="application">
```
Comment (anthonydillon) [inline templates/careers/base_job-details.html:8]: I don't think we need the `u-extra-space` class anymore.

File: templates/careers/base_job-details.html [hunk 3]
```diff
@@ -2,8 +2,44 @@
 
 {% block hero %}{% endblock %}
 
-<section class="p-strip u-extra-space">
+<section class="p-strip u-extra-space" id="details">
```
Comment (anthonydillon) [inline templates/careers/base_job-details.html:5]: I don't think we need the `u-extra-space` class anymore.

File: templates/careers/base_job-details.html [hunk 4]
```diff
@@ -2,8 +2,44 @@
...
+<section class="p-strip u-extra-space u-hide" id="application">
+  {% block application_content %}{% endblock %}
+</section>
+
+<script>
+  const apply_button = document.getElementById("apply-button")
+  const back_roles = document.getElementById("back-roles")
+  const back_details = document.getElementById("back-details")
+  job_details = document.getElementById("details")
+  application_form = document.getElementById("application")
```
Comment (anthonydillon) [inline templates/careers/base_job-details.html:17]: Missing `const`

File: templates/careers/base_job-details.html [hunk 5]
```diff
@@ -2,8 +2,44 @@
...
+  const origin_url = window.location.pathname
+  
+  apply_button.addEventListener("click", function(){
+    show_application()
+    window.history.replaceState({}, document.title, `${window.location.pathname}/application`);
```
Comment (anthonydillon) [inline templates/careers/base_job-details.html:22]:
```suggestion
    window.history.replaceState({}, document.title, `${origin_url}/application`);
```

File: templates/careers/base_job-details.html [hunk 6]
```diff
@@ -2,8 +2,44 @@
...
+  })
+  
+  back_details.addEventListener("click", function(){
+    hide_application()
+    window.history.replaceState({}, document.title, origin_url);
+  })
+  
+  function show_application(){
```
Comment (anthonydillon) [inline templates/careers/base_job-details.html:30]: Could you also add a scrollTo(application_form) so the user is in the right place?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
A few suggestions in the code

**Comment by marinacastejon** [comment]
Hi @mtruj013 thanks for this. LGTM, except for a small detail. When I go back in the browser from the application form page, it takes me to the open roles page, instead of the previous page which is the job description page. Is it feasible to change the behaviour so that I'm directed to the job description page? Thanks

**Comment by mtruj013** [comment]
@anthonydillon @marinacastejon @lyubomir-popov hi all, thanks for the feedback! I've addressed your comments if you all could take another look?

File: static/sass/styles.scss [hunk 7]
```diff
@@ -203,7 +203,7 @@ $p-small-lh-diff: map-get($line-heights, default-text) -
   }
 
   h3 {
-    @extend %vf-heading-4;
+    @extend %vf-heading-5;
```
Comment (anthonydillon) [inline static/sass/styles.scss:206]: Nice 👍

**Review by anthonydillon — APPROVED** [review]
LGTM :+1:

**Comment by marinacastejon** [comment]
@mtruj013 Thanks for this, LGTM 👍

**Comment by mtruj013** [comment]
@lyubomir-popov

**Pull request merged**